cantools/database/can/c_source.py

```diff
@@ -1350,11 +1365,18 @@ def _generate_declarations(database_name, messages, floating_point_numbers):
             signal_declaration = ''
 
             if floating_point_numbers:
-                signal_declaration = SIGNAL_DECLARATION_ENCODE_DECODE_FMT.format(
-                    database_name=database_name,
-                    message_name=message.snake_name,
-                    signal_name=signal.snake_name,
-                    type_name=signal.type_name)
+                if not node_name or _is_sender(message, node_name):
```

Comment:
Move `not node_name or` into the helper function to make this line `if _is_sender(message, node_name):`

Reply:
The problem is, currently:
1. packing functions  are generated if `_is_sender` 
2. unpacking functions  are generated if `not _is_sender` 
3. both packing and unpacking are generated if `node_name` name isn't provided (None).  

One function will not work to cover the 1,2  and 3. Also, it wouldn't make sense if node was not provided. 

Maybe it would be a bit mode clear if I checked against null instead of `not node_name`:
`_is_sender(message, node_name) or node_name == None` ?

Reply:
Ahh, ok, thne I suggest having two helper functions, `_is_sender()` and `_is_receiver()`. Both returns true if `node_name` is `None`.